pkg/aws/validations/iam_helpers.go

```diff
@@ -62,3 +67,18 @@ func IamResourceHasTag(iamTags []iamtypes.Tag, tagKey string, tagValue string) b
 
 	return false
 }
+
+func IamRoleArnsValidator(cluster *cmv1.Cluster) error {
+	validatingMap := map[string]struct{}{}
+	clusterIamRoles := GetAccountRolesArnsMap(cluster)
+	maps.Copy(clusterIamRoles, GetOperatorRolesArnsMap(cluster))
+
+	for _, arn := range clusterIamRoles {
+		if _, exist := validatingMap[arn]; exist {
+			return fmt.Errorf("ROSA IAM roles requires unique ARNs and should not be shared by other cluster IAM roles. "+
+				"Duplicated role arn: %s", arn)
+		}
+		validatingMap[arn] = struct{}{}
+	}
```

Comment:
I'm wondering if it would be worth it to make the error string a var, to also be used in [this test](https://github.com/openshift-online/ocm-common/pull/61/files#diff-d7eeb1dabfba210209b98f7b475fc3c418b5c0f6302f8d8b4e82f637c82f78c6R167)